README.md

```diff
@@ -0,0 +1,269 @@
+## Generic Bucket Provisioning
+
+Kubernetes natively supports dynamic provisioning for many types of file and
+block storage, but lacks support for object bucket provisioning. 
+This repo is a temporary placeholder for bucket provisioning CRDs and related
+generated client code. The longer term goal is to move this repo to a Kubernetes
+repo such as sig-storage or an external storage repo.
+
+### Design Overview
+S3 compatible object store buckets are stable and consistent among most, if not
+all, object store solutions. Therefore, the time has come where we can
+support a bucket provisioning API similar to that used for Persistent Volumes.
+We propose two new Custom Resources to abstract an object store bucket and
+a claim/request for such a bucket. 
+
+An `ObjectBucketClaim` (OBC) is similar in usage to a Persistent Volume Claim
+(PVC), and an `ObjectBucket` (OB) is the Persistent Volume (PV) equivalent. Only 
+dynamic bucket provisioning will be supported. In other words, a storage admin
+cannot pre-allocate buckets and expect OBCs to bind to them. Bucket binding
+refers to the actual bucket being created by the underlying object store provider.
+An OBC is namespaced and references a storage class which defines the object
+store itself. The details of the object store (ceph, minio, cloud, on-prem)
+are not apparent to the developer and can change without disturbing the app.
+An OB is non-namespaced (global), typically not visible to end users, and will
+contain some info pertinent to the provisioned bucket. Like PVs, there is a 1:1
+binding of an OBC to a OB.
+
+Even though OBCs and OBs are generic, the actual bucket provisioning is done by
+object store specific operators. For example, if the underlying object store is
+AWS S3, the developer creates an OBC, referencing a Storage Class which references
+the S3 store. The cluster has the S3 provisioner running which is watching for
+OBCs that it knows how to handle. Other OBCs are ignored by the S3 provisioner.
+Likewise, the same cluster can also have a rook-ceph RGW provisioner running which also watches OBCs. Like the S3 proivisioner, it only handles OBCs that it knows
+how to provision and skips the rest.
+
+In order to access the bucket, the endpoint and key-pair access keys need to be
+availble to an application pod, and the pod should not run until the bucket has
+been provisioned and can be accessed, This is true even if the pod is created
+prior to the OBC.
+To synchronize pods with buckets, a ConfigMap will contain the bucket endpoint
+and a secret will contain the key-pair credentials. Pods already block waiting
```

Comment:
ConfigMap is capitalized, secret is not.

Reply:
> Pods already block waiting for secrets and config maps to be mounted, so we have simple, familiar synronization.

- "block waiting" - i think this should be either "block" or "waiting"
- "synronization" typo